core/src/main/java/org/apache/gravitino/connector/BaseCatalog.java

```diff
@@ -181,53 +182,59 @@ public CatalogOperations ops() {
 
   public AuthorizationPlugin getAuthorizationPlugin() {
     if (authorization == null) {
-      synchronized (this) {
-        if (authorization == null) {
-          BaseAuthorization<?> baseAuthorization = createAuthorizationPluginInstance();
-          if (baseAuthorization == null) {
-            return null;
-          }
-          authorization = baseAuthorization;
-        }
-      }
+      return null;
     }
     return authorization.plugin(provider(), this.conf);
   }
 
-  private BaseAuthorization<?> createAuthorizationPluginInstance() {
+  public void initAuthorizationPluginInstance(IsolatedClassLoader classLoader) {
+    if (authorization != null) {
+      return;
+    }
+
     String authorizationProvider =
         catalogPropertiesMetadata().containsProperty(AUTHORIZATION_PROVIDER)
             ? (String) catalogPropertiesMetadata().getOrDefault(conf, AUTHORIZATION_PROVIDER)
             : null;
 
     if (authorizationProvider == null) {
       LOG.info("Authorization provider is not set!");
-      return null;
+      return;
     }
 
-    ServiceLoader<AuthorizationProvider> loader =
-        ServiceLoader.load(
-            AuthorizationProvider.class, Thread.currentThread().getContextClassLoader());
-
-    List<Class<? extends AuthorizationProvider>> providers =
-        Streams.stream(loader.iterator())
-            .filter(p -> p.shortName().equalsIgnoreCase(authorizationProvider))
-            .map(AuthorizationProvider::getClass)
-            .collect(Collectors.toList());
-    if (providers.isEmpty()) {
-      throw new IllegalArgumentException(
-          "No authorization provider found for: " + authorizationProvider);
-    } else if (providers.size() > 1) {
-      throw new IllegalArgumentException(
-          "Multiple authorization providers found for: " + authorizationProvider);
-    }
+    BaseAuthorization<?> baseAuthorization;
     try {
-      return (BaseAuthorization<?>)
-          Iterables.getOnlyElement(providers).getDeclaredConstructor().newInstance();
+      baseAuthorization =
```

Comment:
Why not assign it directly to authorization?

Reply:
Because `private volatile BaseAuthorization<?> authorization;`
Just like `CatalogManager::createCatalogInstance()` to create a `BaseCatalog`.

Reply:
I could not it clearly, as `authorization` is marked as `volatile`, so we can't assign value to it directly?

Reply:
Fix.